src/templatizer/array-selector.html

```diff
@@ -158,58 +187,48 @@
       }
 
       _applySplices(splices) {
+        let toRemove = [];
```

Comment:
Need a comment here to explain a fundamental limitation of this approach: if an array is re-ordered the same element may be removed and re-added. In this case, it may not be detected as still being in the selection and therefore the selection will not be maintained in that case.